packages/main/src/ColorPalette.js

```diff
@@ -209,16 +254,131 @@ class ColorPalette extends UI5Element {
 	}
 
 	_onkeyup(event) {
-		if (isSpace(event)) {
+		if (isSpace(event) && event.target.localName === "ui5-color-palette-item") {
 			event.preventDefault();
 			this.selectColor(event.target);
 		}
 	}
 
 	_onkeydown(event) {
-		if (isEnter(event)) {
+		if (isEnter(event) && event.target.localName === "ui5-color-palette-item") {
+			this.selectColor(event.target);
+		}
+	}
+
+	_onDefaultColorKeyDown(event) {
+		if (isTabNext(event) && this.popupMode) {
+			event.preventDefault();
+			this._onDefaultColorClick();
+		}
+
+		if (isDown(event)) {
+			event.stopPropagation();
+
+			this.focusColorPaletteElement(this.colorPaletteNavigationElements[1]);
+		} else if (isUp(event)) {
+			event.stopPropagation();
+			const hasRecentColors = this.showRecentColors && this.recentColors[0];
```

Comment:
Instead of declaring this in every function, you can put it in a getter which you can use where necessary.